### PR TITLE
rsx/vk: Optimizations

### PR DESCRIPTION
**Pull request by kd-11**
Applies some optimizations to RSX (shared) and Vulkan.
- Refactors some large RSX headers to be easier to work with. This started to be a problem during implementation of this task so I took the opportunity to handle it here.
- Avoids processing unnecessary vertex streams for draw batches. Given that we have several tens of thousands of calls of this method in one frame in some titles, even small saves can stack up quite a bit.
- Avoid calling get_system_time() when we do not need to know the actual time/timediff. A monotonic counter is already provided by rsx::get_shared_tag() for the purposes of determining if events occurred earlier or later relative to each other and is much cheaper to use.
- Avoids using std::this_thread::thread_id() which has unexpectedly high cost on some platforms. thread_ctrl provides most of what we need using tls to determine thread identity.
- Batch descriptor allocation requests instead of making one allocation request for each drawcall. Now up to 64 descriptors are allocated in one go. The cost of allocating one descriptor is virtually the same as all 64.
- Enable use of VK_EXT_descriptor_indexing extension to update bound descriptors in a deferred manner. Updating thousands of descriptors in one go is orders of magnitude faster than updating each descriptor individually. Wins here are limited by hardware and drivers, with the batch sizes determined by descriptor usage and update-after-bind support provided by the hw/driver combo.

With this set of changes up to 30% more performance can be observed in purely RSX-constrained situations. The optimizations improve the speed of command recording and not command execution. As long as command execution takes longer than command recording (high end CPU paired with low end GPU) only a minor speedup will be observed. I still have more optimizations in the works, but I decided to get it out in smaller batches for ease of integration.

NOTE: The shader cache itself does not have altered structure, but some fields have been removed internally which will trigger a shader cache rebuild. Some more drastic changes are expected on that front, so I refrained from bumping the cache version for now.

### CODE REVIEW

File: rpcs3/emucore.vcxproj.filters [hunk 1]
```diff
@@ -1991,6 +1991,21 @@
       <Filter>Emu</Filter>
     </ClInclude>
   </ItemGroup>
+  <ItemGroup>
+    <None Include="Emu\RSX\Common\Interpreter\FragmentInterpreter.glsl">
+    <ClInclude Include="Emu\Audio\audio_device_listener.h">
```
Comment (Megamouse) [inline rpcs3/emucore.vcxproj.filters:1996]: I think this is already present in the filters

**Comment by DefaltBR** [comment]
Improvements indeed! Uncharted 2 was the perfect test for me, because it was, somehow, GPU intensive when i used any resolution scale above 100% (5600x & RTX 2060 Super)... And i saw improvements right from the main menu! TLoU also had 1~2 fps more...

PR
UC2_PR

Master
UC2_Master

[PR.log.gz](https://github.com/RPCS3/rpcs3/files/7235963/PR.log.gz)
[Master.log.gz](https://github.com/RPCS3/rpcs3/files/7235965/Master.log.gz)

File: rpcs3/emucore.vcxproj.filters (hunk 1 above)
Comment (kd-11) [reply rpcs3/emucore.vcxproj.filters:1996]: Indeed, this is likely an issue with msvc editing the files + rebasing the branch. Will fix.

**Comment by MsDarkLow** [comment]
8700k @ 4.8GHz
Yakuza Kenzan - Test: 36.2 fps -> 44.8 fps
Yakuza Dead Souls - Test 1: 73.5 fps -> 77.5 fps; 
Yakuza Dead Souls - Test 2: 88.7 fps -> 91.0 fps:

Yakuza Ishin spots are harder to tell since there is a lot of variance per boot but it may still be worth noting.
Test 1: 159.4 fps -> 168.6 fps
Test 2: 31.0 fps -> 32.9 fps

<details>
<summary>Yakuza Kenzan</summary>

Master
KenzanMaster

PR
KenzanPR
</details>

<details>
<summary>Yakuza Dead Souls</summary>

Master
DS2Master
DS1Master

PR
DS2PR
DS1PR
</details>

<details>
<summary>Yakuza Ishin</summary>

Master
Ishin1Master
Ishin2Master

PR
Ishin1PR
Ishin2PR

</details>

**Comment by cipherxof** [comment]
Master:
master

PR:
pr

**Comment by Jonathan44062** [comment]
Ratchet and Clank Collection have some improves in this PR
CPU: i7 8700

Ratchet & Clank 1
Master
R C1 Master

PR
R C1 PR

Ratchet & Clank: Going Commando
Master
R C2 Master

PR
R C2 PR

Ratchet & Clank: Up Your Arsenal
Master
R C3 Master

PR
R C3 PR

**Comment by DefaltBR** [comment]
Found an issue today... In GoW3, specifically in Poseidon's Chamber, as you can see below (WCB is on), there's a strong brightness fading in and out repeatedly... Tried with previous build, same settings, and it doesn't happen...

After this PR
GoW3_Actual_Master

Before this PR
GoW3_Previous_Master
[Current_master.log.gz](https://github.com/RPCS3/rpcs3/files/7246973/Current_master.log.gz)
[Previous_master.log.gz](https://github.com/RPCS3/rpcs3/files/7246975/Previous_master.log.gz)

**Pull request merged**